### PR TITLE
Add Svelte 5 to the allowed peer dependency range

### PR DESCRIPTION
**Pull request by lendle**
Svelte 5 is "almost" entirely backwards compatible with svelte 4. Breaking changes can be found [here](https://svelte.dev/docs/svelte/v5-migration-guide#Breaking-changes-in-runes-mode). 

`@xstate/svelte` only uses `onDestroy`, `readable`, and `Readable` from svelte, which are backwards compatible.

### CODE REVIEW

**Comment by lendle** [comment]
There are some failing svelte related tests, but I think this is coincidental, they're also failing in main. I created the branch from the last working commit on main (65088b6a595a0c69f312fec5f91916e1dbc81b46) and it works locally. So I think the ci build will succeed once main is fixed.

**Comment by lendle** [comment]
Build seems to be working now if anyone is available to review. cc @Andarist

File: .changeset/long-ravens-yell.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@xstate/svelte': patch
+---
+
+Bump svelte version to include svelte 5
```
Comment (Andarist) [inline .changeset/long-ravens-yell.md:5]:
```suggestion
Add Svelte 5 to the allowed peer dependency range
```

**Comment by lendle** [comment]
🙇 Thanks!

**Pull request merged**